rdr_service/dao/participant_summary_dao.py

```diff
@@ -894,6 +928,98 @@ def add_to_sqlalchemy_query(self, query, field):
+                    ParticipantSummary.consentCohort == ParticipantCohort.COHORT_1,
+                    ParticipantSummary.consentForGenomicsRORAuthored > eighteen_month_ago
+                ),
+                and_(
+                    ParticipantSummary.consentCohort == ParticipantCohort.COHORT_2,
+                    ParticipantSummary.consentForGenomicsRORAuthored > eighteen_month_ago
+                )
+            )
+            passive_criterion = and_(
+                or_(
+                    ParticipantSummary.questionnaireOnHealthcareAccessAuthored == None,
+                    ParticipantSummary.questionnaireOnHealthcareAccessAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnFamilyHealthAuthored == None,
+                    ParticipantSummary.questionnaireOnFamilyHealthAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnMedicalHistoryAuthored == None,
+                    ParticipantSummary.questionnaireOnMedicalHistoryAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnCopeOctAuthored == None,
+                    ParticipantSummary.questionnaireOnCopeOctAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnCopeJulyAuthored == None,
+                    ParticipantSummary.questionnaireOnCopeJulyAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnCopeJuneAuthored == None,
+                    ParticipantSummary.questionnaireOnCopeJuneAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.questionnaireOnCopeMayAuthored == None,
+                    ParticipantSummary.questionnaireOnCopeMayAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.consentCohort != ParticipantCohort.COHORT_1,
+                    ParticipantSummary.consentForStudyEnrollmentAuthored ==
+                    ParticipantSummary.consentForStudyEnrollmentFirstYesAuthored,
+                    ParticipantSummary.consentForStudyEnrollmentAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.consentCohort != ParticipantCohort.COHORT_1,
+                    ParticipantSummary.consentForGenomicsRORAuthored == None,
+                    ParticipantSummary.consentForGenomicsRORAuthored <= eighteen_month_ago
+                ),
+                or_(
+                    ParticipantSummary.consentCohort != ParticipantCohort.COHORT_2,
+                    ParticipantSummary.consentForGenomicsRORAuthored == None,
+                    ParticipantSummary.consentForGenomicsRORAuthored <= eighteen_month_ago
+                )
```

Comment:
Not a critical thing, but are these checks of "not having authored date OR having it be more than eighteen months ago" to make sure that a participant isn't ACTIVE? Would it work to have the passive filter below be set up as "passive_criterion AND NOT active_criterion"?

Reply:
Yeah, that's what I did before, but it doesn't work, the result is not what we expected. I didn't find the reason yet.

Reply:
I think I see why that is. Any comparison of NULL to something results in NULL. So `SELECT NULL <= 8` and `SELECT NOT(NULL > 8)` both give NULL, which is falsy. Coalesces in the active_criterion would fix that.

Reply:
Ah, yes, I think that the reason.